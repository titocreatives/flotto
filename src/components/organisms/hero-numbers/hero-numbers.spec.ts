import { shallowMount } from "@vue/test-utils";
import HeroNumbers from "./";

const MOCK_NUMBERS: string[] = ["25", "31", "38", "49", "50"];
const MOCK_ADDTIONAL_NUMBERS: string[] = ["5", "10"];
const MOCK_ALL_NUMBERS: string[] = [...MOCK_NUMBERS, ...MOCK_ADDTIONAL_NUMBERS];

describe("Organism HeroNumbers", () => {
  let cmp: any;

  beforeAll(() => {
    cmp = shallowMount(HeroNumbers, {
      propsData: {
        numbers: MOCK_NUMBERS,
        additionalNumbers: MOCK_ADDTIONAL_NUMBERS
      }
    });
  });

  it("matches snapshot", () => {
    expect(cmp.element).toMatchSnapshot();
  });

  it("has correct amount of balls", () => {
    let balls = cmp.findAll(".ball");
    expect(balls).toHaveLength(MOCK_ALL_NUMBERS.length);
  });

  it("renders correct numbers in order", () => {
    let balls = cmp.findAll(".ball-number");

    for (let i = 0; i < balls.length; i++) {
      expect(balls.at(i).text()).toMatch(MOCK_ALL_NUMBERS[i]);
    }
  });

  it("additional numbers have .as-additional modifier class", () => {
    let balls = cmp.findAll(".ball");
    expect(balls.at(5).classes() && balls.at(6).classes()).toContain(
      "as-additional"
    );
  });
});
