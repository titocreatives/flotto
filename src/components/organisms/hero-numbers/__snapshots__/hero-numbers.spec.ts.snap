// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organism HeroNumbers matches snapshot 1`] = `
<div
  class="o hero-numbers"
>
  <div
    class="hero-numbers-grid"
  >
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-default"
      >
        <span
          class="ball-number f-number"
        >
          25
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-default"
      >
        <span
          class="ball-number f-number"
        >
          31
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-default"
      >
        <span
          class="ball-number f-number"
        >
          38
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-default"
      >
        <span
          class="ball-number f-number"
        >
          49
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-default"
      >
        <span
          class="ball-number f-number"
        >
          50
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-additional"
      >
        <span
          class="ball-number f-number"
        >
          5
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
    <div
      class="hero-numbers-cell"
    >
      <figure
        class="ball as-additional"
      >
        <span
          class="ball-number f-number"
        >
          10
        </span>
         
        <span
          class="ball-shadow"
        />
      </figure>
    </div>
  </div>
</div>
`;
